examples/java-matter-controller/java/src/com/matter/controller/commands/common/Command.java

```diff
@@ -0,0 +1,292 @@
+   * @param out A pointer to an Optional where the argv value will be stored, the argv value could
+   *     be empty
+   * @param desc The description of the argument that will be displayed in the command help
+   * @return The number of arguments currently added to the command
+   */
+  public final int addArgument(String name, long min, long max, Optional<Long> out, String desc) {
+    Argument arg = new Argument();
+    arg.type = ArgumentType.Number_int64;
+    arg.name = name;
+    arg.value = (Object) out;
+    arg.min = min;
+    arg.max = max;
+    arg.flags = Argument.kOptional;
+    arg.desc = Optional.ofNullable(desc);
+
+    mArgs.add(arg);
+    return mArgs.size();
+  }
+
+  /**
+   * @brief Add an optional short integer command argument
+   * @param name The name that will be displayed in the command help
+   * @param min The minimum value of the argv value
+   * @param max The minimum value of the argv value
+   * @param out A pointer to an Optional where the argv value will be stored, the argv value could
+   *     be empty
+   * @param desc The description of the argument that will be displayed in the command help
+   * @return The number of arguments currently added to the command
+   */
+  public final int addArgument(
+      String name, short min, short max, Optional<Short> out, String desc) {
+    Argument arg = new Argument();
+    arg.type = ArgumentType.Number_int16;
+    arg.name = name;
+    arg.value = (Object) out;
+    arg.min = min;
+    arg.max = max;
+    arg.flags = Argument.kOptional;
+    arg.desc = Optional.ofNullable(desc);
+
+    mArgs.add(arg);
+    return mArgs.size();
+  }
+
+  /**
+   * @brief Add an optional boolean command argument
+   * @param name The name that will be displayed in the command help
+   * @param out A pointer to an Optional where the argv value will be stored, the argv value could
+   *     be empty
+   * @param desc The description of the argument that will be displayed in the command help
+   * @return The number of arguments currently added to the command
+   */
+  public final int addArgument(String name, Optional<Boolean> out, String desc) {
+    Argument arg = new Argument();
+    arg.type = ArgumentType.Bool;
+    arg.name = name;
+    arg.value = (Object) out;
+    arg.flags = Argument.kOptional;
+    arg.desc = Optional.ofNullable(desc);
+
+    mArgs.add(arg);
+    return mArgs.size();
+  }
+
+  public final void initArguments(int argc, String[] args) throws IllegalArgumentException {
+    int argsCount = mArgs.size();
+
+    if (argsCount != argc) {
+      throw new IllegalArgumentException(
+          "Wrong arguments number: " + argc + " instead of " + argsCount);
+    }
+
+    for (int i = 0; i < argsCount; i++) {
+      initArgument(i, args[i]);
+    }
+  }
+
+  public void initArgument(int argIndex, String argValue) throws IllegalArgumentException {
+    boolean isValidArgument = false;
+
+    Argument arg = mArgs.get(argIndex);
+    switch (arg.type) {
```

Comment:
Should this be in the `Argument` class? Something like `mArgs.get(argIndex).setValue(argValue)` (where `setValue` can throw the `IllegalArgumentException` on invalid input). Since the `Argument` class already has knowledge of how it's configured (type, ranges, etc.) it could also be responsible for validation and reassignment.

It may also be considering a hierarchy of `Argument` types, e.g. 

```
class NumericArgument extends Argument
class InetAddressArgument extends Argument
```